src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -16,170 +15,157 @@
+    SPELL_RAISE_DEAD    = 17473, // Inits. 17698 and 17471
+    SPELL_UNHOLY_AURA   = 17467
 };
 
-// Define Add positions
-Position const ADD_POS_1 = {4017.403809f, -3339.703369f, 115.057655f, 5.487860f};
-Position const ADD_POS_2 = {4013.189209f, -3351.808350f, 115.052254f, 0.134280f};
-Position const ADD_POS_3 = {4017.738037f, -3363.478016f, 115.057274f, 0.723313f};
-Position const ADD_POS_4 = {4048.877197f, -3363.223633f, 115.054253f, 3.627735f};
-Position const ADD_POS_5 = {4051.777588f, -3350.893311f, 115.055351f, 3.066176f};
-Position const ADD_POS_6 = {4048.375977f, -3339.966309f, 115.055222f, 2.457497f};
+enum BaronRivendareEvents
+{
+    EVENT_SHADOWBOLT    = 1,
+    EVENT_CLEAVE        = 2,
+    EVENT_MORTALSTRIKE  = 3,
+    EVENT_RAISE_DEAD    = 4
+};
 
-class boss_baron_rivendare : public CreatureScript
+// Raise Dead
+uint32 const RaiseDeadSpells[6] =
+{
+    17475, 17476, 17477,
+    17478, 17479, 17480
+};
+
+struct boss_baron_rivendare : public BossAI
 {
 public:
-    boss_baron_rivendare() : CreatureScript("boss_baron_rivendare") { }
+    boss_baron_rivendare(Creature* creature) : BossAI(creature, TYPE_BARON), RaiseDead(false) { }
 
-    CreatureAI* GetAI(Creature* creature) const override
+    void Reset() override
     {
-        return GetStratholmeAI<boss_baron_rivendareAI>(creature);
+        // DoCastSelf doesn't add aura after reset
+        me->AddAura(SPELL_UNHOLY_AURA, me);
+
+        // needed until re-write of instance scripts is done
+        if (instance->GetData(TYPE_RAMSTEIN) == DONE)
+            instance->SetData(TYPE_BARON, NOT_STARTED);
+
+        BossAI::Reset();
     }
 
-    struct boss_baron_rivendareAI : public ScriptedAI
+    void JustEngagedWith(Unit* who) override
     {
-        boss_baron_rivendareAI(Creature* creature) : ScriptedAI(creature)
-        {
-            Initialize();
-            instance = me->GetInstanceScript();
-        }
+        // needed until re-write of instance scripts is done
+        if (instance->GetData(TYPE_BARON) == NOT_STARTED)
+            instance->SetData(TYPE_BARON, IN_PROGRESS);
 
-        void Initialize()
-        {
-            ShadowBolt_Timer = 5000;
-            Cleave_Timer = 8000;
-            MortalStrike_Timer = 12000;
-            //        RaiseDead_Timer = 30000;
-            SummonSkeletons_Timer = 34000;
-        }
+        events.ScheduleEvent(EVENT_SHADOWBOLT, 5s);
+        events.ScheduleEvent(EVENT_CLEAVE, 8s);
+        events.ScheduleEvent(EVENT_MORTALSTRIKE, 12s);
+        events.ScheduleEvent(EVENT_RAISE_DEAD, 15s);
 
-        InstanceScript* instance;
+        BossAI::JustEngagedWith(who);
+    }
 
-        uint32 ShadowBolt_Timer;
-        uint32 Cleave_Timer;
-        uint32 MortalStrike_Timer;
-        //    uint32 RaiseDead_Timer;
-        uint32 SummonSkeletons_Timer;
+    void JustDied(Unit* killer) override
+    {
+        // needed until re-write of instance scripts is done
+        instance->SetData(TYPE_BARON, DONE);
```

Comment:
it cant be done inside of InstanceScript?
Call/Handle wherever you want when change boss state?

Reply:
@Nitair : Please post a reply to this question.

Reply:
@Keader @illfated

as written https://github.com/TrinityCore/TrinityCore/pull/21687 it cant be done until the instance script is re-written. I'm already on it but it needs some time until creating the PR here (as written https://github.com/TrinityCore/TrinityCore/pull/22572#issuecomment-480497510)